src/audio/mixer.c

```diff
@@ -65,69 +65,164 @@ static void mix_n_s16(struct comp_dev *dev, struct audio_stream *sink,
 		      const struct audio_stream **sources, uint32_t num_sources,
 		      uint32_t frames)
 {
-	int16_t *src;
```

Comment:
Is the saturation issue fixed as a side effect of the optimization access? Otherwise, maybe it worths adding the fix a separate patch.

Reply:
I don't see point in doing separate patch. I would just duplicate with minor change one poor performing funcion.